src/dht/dht_server.cc

```diff
@@ -743,6 +743,11 @@ DhtServer::event_read() {
       if (!message[key_t].is_raw_string())
         throw dht_error(dht_error_protocol, "No transaction ID");
 
+	  //Check transaction id length to avoid use of it in dht_error if invalid
+	  if(message[key_t].as_string().size() > 20) {
+		  throw network_error();
+	  }
```

Comment:
Please try to follow the code style, e.g. use spaces, not tabs. Properly format the comment with a leading space and punctuation.

Use 'as_raw_string().size()' as that doesn't require a conversion to std::string.

Also use dht_error_protocol, not network_error.